platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/JUnitXmlResultWriter.java

```diff
@@ -60,7 +62,7 @@ public void write(TestClassResult result, OutputStream output) {
                 .attribute("failures", String.valueOf(result.getFailuresCount()))
                 .attribute("errors", "0")
 
-                .attribute("timestamp", DateUtils.format(result.getStartTime(), DateUtils.ISO8601_DATETIME_PATTERN))
+                .attribute("timestamp", ISO_INSTANT.format(Instant.ofEpochMilli(result.getStartTime())))
```

Comment:
Are we sure this format with the timezone is well-supported? It seems the original implementation from Ant generated without a timezone.

I'm not really sure what even reads this format so I don't know where to check for reassurance it's accepted.

Reply:
I had the same concern. I looked at several open source JUnit XML parsers across Javascript, Python and Rust as well as Jenkins' parser: https://github.com/gradle/gradle/issues/31841#issuecomment-2581446901

One of the Javascript parsers supported this format. All of the others treat it as a string and don't try to parse it. Jenkins explicitly adds a Z if it's missing. 

pytest started adding timezone in 2024: https://github.com/pytest-dev/pytest/pull/12491

Reply:
I think that's good enough then